controller/rest/rest.go

```diff
@@ -1152,7 +1153,15 @@ func getAgentWorkloadFromFilter(filters []restFieldFilter, acc *access.AccessCon
 func initDefaultRegistries() {
 	// all on lower-case
 	dockerRegistries = utils.NewSet("https://docker.io/", "https://index.docker.io/", "https://registry.hub.docker.com/", "https://registry-1.docker.io/")
-	defaultRegistries = utils.NewSet("https://docker.io/", "https://index.docker.io/", "https://registry.hub.docker.com/", "https://registry-1.docker.io/")
+
+	// if a flag was provided for search registries, use those,
+	// otherwise use the default docker registries
+	if len(searchRegistries.ToSlice()) > 0 {
+		defaultRegistries = searchRegistries
```

Comment:
We also considered this, but it occurred to us that in environments where the unqualified search registry is not `docker.io`, it should probably be omitted altogether. And in the case that `docker.io` is really needed as an additional registry, it can be added to the search registries parameter.

I think this offers more flexibility, as it allows omitting the `docker.io` registry, or changing the order in which it appears in the slice using a single parameter.